backend/pkg/api/types/data_access.go

```diff
@@ -331,3 +352,169 @@ const CtxUserIdKey CtxKey = "user_id"
 const CtxIsMockedKey CtxKey = "is_mocked"
 const CtxMockSeedKey CtxKey = "mock_seed"
 const CtxDashboardIdKey CtxKey = "dashboard_id"
+
+// -------------------------
+// Search structs
+
+// -- General
+type SearchType int
+
+// all possible search types
+const (
+	SearchTypeInteger SearchType = iota
+	SearchTypeName
+	SearchTypeEthereumAddress
+	SearchTypeWithdrawalCredential
+	SearchTypeEnsName
+	SearchTypeGraffiti
+	SearchTypeEmail
+	SearchTypePassword
+	SearchTypeEmailUserToken
+	SearchTypeJsonContentType
+	// Validator Dashboard
+	SearchTypeValidatorDashboardPublicId
+	SearchTypeValidatorPublicKeyWithPrefix
+	SearchTypeValidatorPublicKey
+)
+
+type Searchable interface {
+	GetSearches() []SearchType
+	SetSearchValue(s string)
+	SetSearchType(st SearchType, b bool)
+	IsEnabled() bool
+	HasAnyMatches() bool
+}
+
+// not to be used directly, only for embedding
+// think of this as a base class which provides default implementations that can be overridden / shadowed / masked (see below)
+type basicSearch struct {
+	types map[SearchType]bool
+	value string
+}
+
+func (bs *basicSearch) SetSearchValue(s string) {
+	if bs == nil {
+		log.Warnf("BasicSearch is nil, can't apply search: %s", s)
+		return
+	}
+	bs.value = s
+}
+
+func (bs *basicSearch) SetSearchType(st SearchType, b bool) {
+	if bs.types == nil {
+		bs.types = make(map[SearchType]bool)
+	}
+	bs.types[st] = b
+}
+
+func (bs *basicSearch) IsEnabled() bool {
+	return bs != nil && bs.value != ""
+}
+
+func (bs *basicSearch) HasAnyMatches() bool {
+	for _, v := range bs.types {
+		if v {
+			return true
+		}
+	}
+	return false
+}
+
+func (bs *basicSearch) GetSearches() []SearchType {
+	return []SearchType{
+		SearchTypeName,
+		SearchTypeInteger,
+		SearchTypeEthereumAddress,
+		SearchTypeWithdrawalCredential,
+		SearchTypeEnsName,
+		SearchTypeGraffiti,
+		SearchTypeEmail,
+		SearchTypePassword,
+		SearchTypeEmailUserToken,
+		SearchTypeJsonContentType,
+		SearchTypeValidatorDashboardPublicId,
+		SearchTypeValidatorPublicKeyWithPrefix,
+		SearchTypeValidatorPublicKey,
+	}
+}
+
+type baseSearchResult[T any] struct {
+	Enabled bool
+	Value   T
+}
+
+type SearchNumber baseSearchResult[uint64]
+type SearchString baseSearchResult[string]
+
+func (bs *basicSearch) AsNumber(st SearchType) SearchNumber {
```

Comment:
Don't get why this needs to have a `SearchType` argument, feels like it unnecessarily complicates the API of using searchtypes.

Reply:
It was meant to enable increased but unified flexibility in case that different types should be returned in different formats, e.g. to scale ETH -> wei values or something. Atm I also can't think of good examples so it's true that this looks like unneeded overhead, didn't think it's a big issue as this is encapsulated away from every-day data access workflow. But I'll disable it for now until a real usecase comes up